### PR TITLE
force change changeset versions during release process

### PR DESCRIPTION
**Pull request by angusbayley**
- Automatically upgrade `patch` changesets to `minor` in releases
- Automatically downgrade `minor` and `major` changesets to `patch` in hotfixes

The `changeset` cli itself doesn't provide any ability to modify changeset levels, so this is done via a new action that I've written.

### CODE REVIEW

File: .github/workflows/release-create.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-24.04
     env:
       NODE_OPTIONS: "--max-old-space-size=7168"
-      RELEASE_BRANCH: "release"
+      RELEASE_BRANCH: "release-angusbayley-TEST"
```
Comment (angusbayley) [inline .github/workflows/release-create.yml:12]: to revert

File: .github/workflows/release-create.yml [hunk 2]
```diff
@@ -57,6 +66,6 @@ jobs:
       - name: push
         run: |
           git push origin $RELEASE_BRANCH
-          gh pr create --title ":rocket: Release ${{ steps.date.outputs.date }}" -F .github/templates/release.md --base main --head $RELEASE_BRANCH
+          gh pr create --title ":rocket: [NOT A REAL RELEASE - TEST ONLY, by @angusbayley] Release ${{ steps.date.outputs.date }}" -F .github/templates/release.md --base main --head $RELEASE_BRANCH
```
Comment (angusbayley) [inline .github/workflows/release-create.yml:69]: to revert before merge

File: tools/actions/composites/adjust-changeset-level/action.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        if [[ ! " ${valid_levels[@]} " =~ " ${inputs.from_level} " ]]; then
+          echo "Invalid from_level: ${inputs.from_level}. Must be one of ${valid_levels[@]}"
+          exit 1
+        fi
+        if [[ ! " ${valid_levels[@]} " =~ " ${inputs.to_level} " ]]; then
+          echo "Invalid to_level: ${inputs.to_level}. Must be one of ${valid_levels[@]}"
+          exit 1
+        fi
+
+    - name: Adjust changeset level
```
Comment (angusbayley) [inline tools/actions/composites/adjust-changeset-level/action.yml:26]: Inviting comments on this. As you can see this is done by direct manipulation of the changeset files via `sed` in a shell script. `sed` targets the YAML front matter specifically and not the rest of the file, but this approach still makes the assumption that the format of the YAML front matter in the changeset file will not change to use the `level` in some different context. I think this is a very unlikely scenario and that this approach is fine for a solution that is not intended to last many years.

An alternative to is to do this in node, convert the YAML front matter into JS objects via a standard yaml parser and writer - this is the way the changesets library itself manipulates its changeset files and is more future-proof, but is slightly more code.

File: tools/actions/composites/adjust-changeset-level/action.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      run: echo "Adjusting changeset level from ${{ inputs.from_level }} to ${{ inputs.to_level }}"
+        for file in .changeset/*.md; do
```
[the remark this answers is not in the document]
Comment (angusbayley) [reply tools/actions/composites/adjust-changeset-level/action.yml:28]: yep, I will do this and link to the workflow when done 👍

[the remark this answers is not in the document]
Comment (angusbayley) [reply tools/actions/composites/adjust-changeset-level/action.yml:31]: [Here is a new PR](https://github.com/LedgerHQ/ledger-live/pull/8449/files) based off this one, which adds a patch changeset that should be updated by the changes in this PR when the release workflow runs.

[Here is the release create branch workflow run](https://github.com/LedgerHQ/ledger-live/actions/runs/11976088855/job/33390956845#step:16:119) for that PR. A the line I have linked, you can see that the mentions of "patch" in the description are unchanged, while the level of the changeset has been upgraded from patch to minor.

[the remark this answers is not in the document]
Comment (themooneer) [reply tools/actions/composites/adjust-changeset-level/action.yml:31]: thanks ! sounds good for me

**Review by themooneer — APPROVED** [review]
LGTM

File: .github/workflows/release-create.yml (hunk 1 above)
Comment (angusbayley) [reply .github/workflows/release-create.yml]: done

File: .github/workflows/release-create.yml (hunk 2 above)
Comment (angusbayley) [reply .github/workflows/release-create.yml]: done

**Pull request merged**